src/about_equality_objects.rb

```diff
@@ -0,0 +1,77 @@
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+class AboutEqualityObjects < Neo::Koan
+
+def test_equality_at_object_level_returns_true_only_if_are_the_same_object
+	a = Object.new
+	b = Object.new
+	assert_equal __(false), a == b
+	assert_equal __(true), a == a
+	assert_equal __(false), a === b
+	assert_equal __(false), a.equal?(b)
+	assert_equal __(true), a.eql?(a)
+end
+
+def test_equality_method_is_overriden_in_descendant_classes_to_provide_class_specific_meaning
+	assert_equal __(true), "equality" == "equality"
+	assert_equal __(true), 16 == 16
+end
+
+# Note that the left side String is a different object than the right side String, 
+# but the result of comparing them it turns out true. Why is that? Because descendant
+# classes of BasicObject, like String class, override this methods (==, ===, eql?) to
+# provide class specific meaning. The same thing happens with Fixnum. 
+
+def test_eql_method_only_returns_true_if_objects_have_the_same_value
+	hash1 = { :name => "Juan", :first_name => "Lopez" }
+    hash2 = { :first_name => "Lopez", :name => "Juan" }
+	assert_equal __(true), hash1.eql?(hash2)
+	assert_equal __(true), "equality".eql?("equality")
+end
+
+# The eql? method returns true if one object and another object have the same value. 
+# This method is used by Hash to test members for equality neverminding order
+
+def test_equal_method_should_never_be_overriden_by_descendant_classes
+	assert_equal __(true), "string" == "string"
```

Comment:
I think this is already covered in "test_equality_method_is_overriden_in_descendant_classes_to_provide_class_specific_meaning". I also think that the next two lines should be in that test. I think the only case that makes sense to show here is ``equal?``, since it looks like you want to show that we mustn't override equal?.

What do you think?